jitbuilder/release/src/StructArray.cpp

```diff
@@ -0,0 +1,181 @@
+/******************************************************************************
+ Copyright IBM Corp. 2016, 2016
+ *
+ *  This program and the accompanying materials are made available
+ *  under the terms of the Eclipse Public License v1.0 and
+ *  Apache License v2.0 which accompanies this distribution.
+ *
+ *      The Eclipse Public License is available at
+ *      http://www.eclipse.org/legal/epl-v10.html
+ *
+ *      The Apache License v2.0 is available at
+ *      http://www.opensource.org/licenses/apache2.0.php
+ *
+ * Contributors:
+ *    Multiple authors (IBM Corp.) - initial implementation and documentation
+ ******************************************************************************/
+
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdint.h>
+#include <dlfcn.h>
+#include <errno.h>
+
+#include "Jit.hpp"
+#include "ilgen/TypeDictionary.hpp"
+#include "ilgen/MethodBuilder.hpp"
+#include "StructArray.hpp"
+
+
+void printStructElems(int8_t type, int32_t value)
+   {
+   #define PRINTSTRUCTELEMS_LINE LINETOSTR(__LINE__)
+   printf("StructType { type = %#x, value = %d }\n", type, value);
+   }
+
+
+StructArrayMethod::StructArrayMethod(TR::TypeDictionary *d)
+   : MethodBuilder(d)
+   {
+   DefineLine(LINETOSTR(__LINE__));
+   DefineFile(__FILE__);
+
+   DefineName("testStructArray");
+
+   StructType = d->LookupStruct("Struct");
+   pStructType = d->PointerTo(StructType);
+
+   DefineFunction((char *)"printStructElems",
+                  (char *)__FILE__,
+                  (char *)PRINTSTRUCTELEMS_LINE,
+                  (void *)&printStructElems,
+                  NoType,
+                  2,
+                  Int8,
+                  Int32);
+   }
+
+bool
+StructArrayMethod::buildIL()
```

Comment:
True, using `malloc` to allocate enough memory would work but that feels a bit too hacky. Also, I wasn't sure that calling `malloc` from the first method would do what I wanted but if it does, then I'll use that approach.